modules/core/include/opencv2/core/types_c.h

```diff
@@ -1356,7 +1356,15 @@ CV_INLINE  CvSlice  cvSlice( int start, int end )
 #if !(defined(CV__ENABLE_C_API_CTORS) && defined(__cplusplus))
```

Comment:
Please replace this line instead:

```
#if defined(CV__ENABLE_C_API_CTORS) && defined(__cplusplus) && !defined(__CUDACC__)
```

(see [line](https://github.com/opencv/opencv/pull/12832/files#diff-117355cf66cd2c3d602b63e3c7c527caR1346) 1346)

Reply:
In my project, the code that your suggests occurs a compile error.

Reply:
Corrected:
```
#if !(defined(CV__ENABLE_C_API_CTORS) && defined(__cplusplus) && !defined(__CUDACC__))
```

Reply:
I confirmed there are no problems on compiling and testing when I changed to your suggested code instead of my code. So I will commit the new code tonight. Thanks. ^^